core/src/main/java/org/eclipse/ec4j/core/model/EditorConfig.java

```diff
@@ -17,80 +17,171 @@
 package org.eclipse.ec4j.core.model;
 
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
 
+import org.eclipse.ec4j.core.EditorConfigConstants;
+import org.eclipse.ec4j.core.ResourcePaths.ResourcePath;
 import org.eclipse.ec4j.core.model.optiontypes.OptionTypeRegistry;
 
 /**
+ * A immutable model of an {@code .editorconfig} file.
+ *
  * @author <a href="mailto:angelo.zerr@gmail.com">Angelo Zerr</a>
  */
 public class EditorConfig {
 
-    private final List<Section> sections;
-    private final OptionTypeRegistry registry;
-    private final String version;
+    /**
+     * {@link EditorConfig} builder.
+     */
+    public static class Builder {
+        final OptionTypeRegistry registry;
 
-    private Boolean root;
-    private String dirPath;
+        ResourcePath resourcePath;
+        Boolean root;
+        List<Section> sections = new ArrayList<>();
```

Comment:
Should be better to initialize sections in the construcor and set sections a final, no?